modules/Bio/EnsEMBL/Compara/RunnableDB/OrthologQM/GeneOrderConservation.pm

```diff
@@ -82,16 +85,21 @@ sub fetch_input {
     print "done!\n" if $self->debug;
         
     print "\tfetching and sorting gene_members... " if $self->debug;
-    my $all_gene_member_ids = $self->_get_all_gene_member_ids_in_homology_file;
-    my $sql = 'SELECT gene_member_id, dnafrag_id, dnafrag_start, dnafrag_strand FROM gene_member WHERE gene_member_id = ?';
-    my $sth = $self->compara_dba->dbc->prepare($sql);
-    my @gene_members_unordered;
-    foreach my $this_gm_id ( @$all_gene_member_ids ) {
-        $sth->execute($this_gm_id);
-        my $results = $sth->fetchall_arrayref;
-        push( @gene_members_unordered, $results->[0] );
+    my $all_gene_member_ids = $self->_get_gene_member_ids_hash_for_homology_file;
+    my @gene_members_ordered;
+    foreach my $gdb (@$genome_dbs) {
+        my $gene_file = $self->param_required('gene_dumps_dir') . '/gene_member.' . $gdb->dbID . ".tsv";
+        open( my $gene_fh, '<', $gene_file) or die "Cannot open $gene_file";
+        my $header = <$gene_fh>;
+        my @head_cols = split(/\s+/, $header);
+        while ( my $line = <$gene_fh> ) {
+            my $row = map_row_to_header( $line, \@head_cols );
+            if ($all_gene_member_ids->{$row->{'gene_member_id'}}) {
+                push @gene_members_ordered, [$row->{'gene_member_id'}, $row->{'dnafrag_id'}, $row->{'dnafrag_start'}, $row->{'dnafrag_strand'}];
+            }
+        }
+        close($gene_fh);
     }
-    my @gene_members_ordered = sort { $a->[1] <=> $b->[1] || $a->[2] <=> $b->[2] } @gene_members_unordered;
```

Comment:
I'm not sure if it matters, but the the previous `@gene_members_ordered` was sorted by `genome_db_id` and this new one is by `dnafrag_id` and `dnafrag_start` (based on the query that generates the files), so they may not be the same unless you are certain the files are read in order sorted by `genome_db_id`.

Reply:
As far as I understand the previous `@gene_members_ordered` was sorted by `dnafrag_id` and then `dnafrag_start`, The new one is still sorted on these two fields, but within each genome_db: the `dnafrag_ids` are not globally ordered. It doesn't matter because the index that is created just below (`neighbourhood`) only needs the genes of each `dnafrag_id` to be ordered by `dnafrag_start` with no assumption how the `dnafrag_ids` themselves are ordered. They could even be interleaved, it would still work

Reply:
Why on earth did I ever sort them using perl instead of SQL?!! 🤯😅